### DISCUSSION TITLE
LDM notes for August 16, 2023

### DISCUSSION THREAD

**Discussion by MadsTorgersen** [dotnet/csharplang#7516] [LDM Notes]
https://github.com/dotnet/csharplang/blob/main/meetings/2023/LDM-2023-08-16.md

## Agenda

- Ref-safety scope for collection expressions
- Experimental attribute

**Comment by En3Tho** [comment]
I wonder if we can combine `scoped` and a collection expressions in Span case? E.g.  `scoped Span<int> span = [1, 2, 3]`.

Or utilize `stackalloc` as in `Span<int> span = stackalloc [1, 2, 3]`. I don't think there should be a conflict because type name isn't used with stackalloc in this case.

**Reply by CyrusNajmabadi** [reply]
Neither of these is necessary.  The collection letter already has local scope. It will already be allocated on the stack in your example (without needing to say 'scoped')

**Reply by jnm2** [reply]
From the notes:
> If we pick global and they want local, in the example of the local variable `s` above, they could explicitly add `scoped` to the variable declaration. However, in other scenarios, e.g. when passing the collection expression as a method argument, there is no place to put `scoped`. They would need to factor it out to a local variable or something to achieve that.

**Reply by En3Tho** [reply]
@jnm2 Oh I completely missed that. Thanks.
